tck/src/main/java/org/eclipse/microprofile/lra/tck/service/spi/LRAManagementService.java

```diff
@@ -0,0 +1,42 @@
+/*
+ *******************************************************************************
+ * Copyright (c) 2020 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+package org.eclipse.microprofile.lra.tck.service.spi;
+
+import java.net.URI;
+
+/**
+ * This interface provides an SPI for the management of the LRAs started and processed
+ * in the TCK. It covers necessary operations that should be provided by the 
+ * implemenation. However, because the implementations can differ in the underlying structures
+ * (e.g., coordination vs decentralization approaches) they cannot be specified directly in the
+ * LRA specification.
+ */
+public interface LRAManagementService {
+
+    /**
+     * Returns a boolean value indicationg whether the LRA is already in the end state
+     * (Cancelled, Closed, FailedToCancel, FailedToClose).
+     * 
+     * @param lraId the LRA context of the queried LRA
+     * @return true if the LRA is in the end state, false otherwise
+     */
+    boolean isLRAFinished(URI lraId);
```

Comment:
do you mean to move the try-enlist code into default implementation? I thought the point was to get rid of it.

Reply:
I mean we should not force implementations to provide this new SPI. The TCK already provides an implementation of this new interface you are proposing so we should keep it (my comment above explains why we should keep the original default implementation).

Reply:
I personally don't like the current implementation of the `isLRAFinished` method as it is. When I was developing some new test (#305) or cleaning some tests (#314) I have issues to understand what's happening (suddenly there are enlisted arbitrary participants which shows warning messages in log). As well the implementation is not bullet proof as it shows the LRA as finished when it was has not been started yet.

I beg for pardon but I don't remember what was the discussion on gitter (per #298) so maybe there was some details that I'm missing.
By me forcing implementation to implement its own `isLRAFinished` method is not an issue. While I think that it would be nice passing as less as possible burden to implementations.
I'm not sure if that's completely possible but what to use the `boolean isLRAFinished(URI lra, LRAMetricService lraMetricService, String resourceName)` instead of simple `isLRAFinished(URI lra)` at all places?